crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -250,20 +251,25 @@ impl SpaceInternalHandler for Space {
                 let sign_req = SignNetworkDataEvt {
                     space: space.clone(),
                     agent: agent.clone(),
-                    data: Arc::new(data),
+                    data: Arc::new(data.clone()),
                 };
                 let sig = evt_sender.sign_network_data(sign_req).await?;
-                let agent_info_signed =
-                    crate::types::agent_store::AgentInfoSigned::try_new(sig, agent_info)?;
+                let agent_info_signed = crate::types::agent_store::AgentInfoSigned::try_new(
+                    (*agent).clone(),
+                    sig,
+                    data,
+                )?;
                 tracing::debug!(?agent_info_signed);
```

Comment:
```suggestion
                tracing::debug!(?agent_info, ?sig);
```

?? - not sure the best thing to debug now, but just the bunch of bytes probably not all that useful